src/writer.rs

```diff
@@ -169,20 +169,134 @@ impl<W: Write> Writer<W> {
-        let name = parent_name.as_ref();
-        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
-        self.write_event(event)?;
-        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
-        Ok(())
+    #[must_use]
+    pub fn create_element<'a, N>(&'a mut self, name: &'a N) -> ElementWriter<W>
+    where
+        N: 'a + AsRef<[u8]> + ?Sized,
+    {
+        ElementWriter {
+            writer: self,
+            start_tag: BytesStart::borrowed_name(name.as_ref()),
+        }
+    }
+}
+
+pub struct ElementWriter<'a, W: Write> {
+    writer: &'a mut Writer<W>,
+    start_tag: BytesStart<'a>,
+}
+
+impl<'a, W: Write> ElementWriter<'a, W> {
+    /// Adds an attribute to this element.
+    pub fn with_attribute<'b, I>(mut self, attr: I) -> Self
+    where
+        I: Into<Attribute<'b>>,
+    {
+        self.start_tag.push_attribute(attr);
+        self
+    }
+
+    /// Add additional attributes to this element using an iterator.
+    ///
+    /// The yielded items must be convertible to [`Attribute`] using `Into`.
+    ///
+    /// [`Attribute`]: attributes/struct.Attributes.html
+    pub fn with_attributes<'b, I>(mut self, attributes: I) -> Self
+    where
+        I: IntoIterator,
+        I::Item: Into<Attribute<'b>>,
+    {
+        self.start_tag.extend_attributes(attributes);
+        self
+    }
+
+    /// Write some text inside the current element.
+    pub fn write_text_content(self, text: BytesText) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event(Event::Start(self.start_tag.to_borrowed()))?;
+        self.writer.write_event(Event::Text(text))?;
+        self.writer
+            .write_event(Event::End(self.start_tag.to_end()))?;
+        Ok(self.writer)
+    }
+
+    /// Write a CData event `<![CDATA[...]]>` inside the current element.
+    pub fn write_cdata_content(self, text: BytesText) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event(Event::Start(self.start_tag.to_borrowed()))?;
+        self.writer.write_event(Event::CData(text))?;
+        self.writer
+            .write_event(Event::End(self.start_tag.to_end()))?;
+        Ok(self.writer)
+    }
+
+    /// Write a processing instruction `<?...?>` inside the current element.
+    pub fn write_pi_content(self, text: BytesText) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event(Event::Start(self.start_tag.to_borrowed()))?;
+        self.writer.write_event(Event::PI(text))?;
+        self.writer
+            .write_event(Event::End(self.start_tag.to_end()))?;
+        Ok(self.writer)
+    }
+
+    /// Write an empty (self-closing) tag.
+    pub fn write_empty(self) -> Result<&'a mut Writer<W>> {
+        self.writer.write_event(Event::Empty(self.start_tag))?;
+        Ok(self.writer)
+    }
+
+    /// Create a new scope for writing XML inside the current element.
+    pub fn write_nested_content<F>(mut self, closure: F) -> Result<&'a mut Writer<W>>
```

Comment:
Would `inner` or `scoped` or any other word make more sense here?

Reply:
`inner` seems better. Not a fan of `scoped` I think.

Reply:
Done